client/src/App.js

```diff
@@ -2,39 +2,70 @@ import React, { useState, useEffect } from 'react';
 import './App.css';
 import Dashboard from './Components/Dashboard/Dashboard';
 import LoginPage from './Components/LoginPage/LoginPage';
+import PrivateRoute from './Components/PrivateRoute/PrivateRoute';
+import {
+  BrowserRouter as Switch,
+  Route,
+  BrowserRouter,
+  Redirect,
+} from 'react-router-dom';
+import { getIdFromLocation } from './helperFunc';
 
 function App() {
-  const [token, setToken] = useState('');
-  const [username, setUsername] = useState('');
-  const [offline, setOffline] = useState(!navigator.onLine);
+  const [token, setToken] = useState(localStorage.getItem('token') || '');
+  const [username, setUsername] = useState(
+    localStorage.getItem('username') || ''
+  );
+
+  /* the user can access the app through two different options:
+  1. indicating username and token manually. In the case of GitHub Enterprise the user also needs to indicate the endpoint.
+  2. Github authentication */
```

Comment:
It's a very common practice to use block comment like this:
```suggestion
  /** 
  * the user can access the app through two different options:
  * 1. indicating username and token manually. In the case of GitHub Enterprise the user also needs to indicate the endpoint.
  * 2. Github authentication 
  **/
```
But this is totally up to you :)